### PR TITLE
Use consistent informer and lister for dependency tracking

### LINKED ISSUE #1835: Figure out if we need knative/eventing#4296
https://github.com/knative/eventing/pull/4296 changes the way trigger dependencies are tracked by switching from the kresource duck to the source duck. Do we need this change too?

### PR DESCRIPTION
**Pull request by Ectelion**
Fixes #1835

## Proposed Changes
Refer to [knative/eventing#4296](https://github.com/knative/eventing/pull/4296) for a relevant change. Same informer/lister construct for tracking dependencies was inherited by the trigger reconciler in the GCP repo.

### CODE REVIEW

File: pkg/apis/broker/v1beta1/test_helper.go [hunk 1]
```diff
@@ -67,31 +67,31 @@ func (t testHelper) AvailableEndpoints() *corev1.Endpoints {
 	}
 }
 
-func (t testHelper) ReadyDependencyStatus() *duckv1.KResource {
-	kr := &duckv1.KResource{}
+func (t testHelper) ReadyDependencyStatus() *duckv1.Source {
+	kr := &duckv1.Source{}
```
Comment (zhongduo) [inline pkg/apis/broker/v1beta1/test_helper.go:71]: nit: we can replace all `kr` in this file with like `src`.

File: pkg/apis/broker/v1beta1/trigger_lifecycle.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (ts *TriggerStatus) MarkDependencyNotConfigured() {
 		"DependencyNotConfigured", "Dependency has not yet been reconciled.")
 }
 
-func (ts *TriggerStatus) PropagateDependencyStatus(ks *duckv1.KResource) {
+func (ts *TriggerStatus) PropagateDependencyStatus(ks *duckv1.Source) {
```
Comment (zhongduo) [inline pkg/apis/broker/v1beta1/trigger_lifecycle.go:144]: same here, replace `ks` and `kc` to something more related to Source.

File: pkg/apis/broker/v1beta1/test_helper.go (hunk 1 above)
Comment (Ectelion) [reply pkg/apis/broker/v1beta1/test_helper.go]: Makes sense, have addressed that.

**Review by danyinggu — COMMENTED** [review]
/lgtm

**Pull request merged**